devicetypes/erocm123/inovelli-2-channel-smart-plug-mcd.src/inovelli-2-channel-smart-plug-mcd.groovy

```diff
@@ -0,0 +1,309 @@
+		// Supervision header is 4 bytes long, two bytes dropped here are the latter two bytes of the supervision header
+		cmd.parameter = cmd.parameter.drop(2)
+		// Updated Command Class/Command now with the remaining bytes
+		cmd.commandClass = cmd.parameter[0]
+		cmd.command = cmd.parameter[1]
+		cmd.parameter = cmd.parameter.drop(2)
+	}
+	log.debug "MultiChannelCmdEncap ${cmd}"
+	def encapsulatedCommand = cmd.encapsulatedCommand([0x32: 3, 0x25: 1, 0x20: 1])
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand, cmd.sourceEndPoint as Integer)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.manufacturerspecificv2.ManufacturerSpecificReport cmd) {
+	log.debug "ManufacturerSpecificReport ${cmd}"
+	def msr = String.format("%04X-%04X-%04X", cmd.manufacturerId, cmd.productTypeId, cmd.productId)
+	log.debug "msr: $msr"
+	updateDataValue("MSR", msr)
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	// This will capture any commands not handled by other instances of zwaveEvent
+	// and is recommended for development so you can see every command the device sends
+	log.debug "Unhandled Event: ${cmd}"
+}
+
+def on() {
+	log.debug "on()"
+	commands([
+		zwave.switchAllV1.switchAllOn(),
+		encap(zwave.switchBinaryV1.switchBinaryGet(), 1),
+		encap(zwave.switchBinaryV1.switchBinaryGet(), 2)
+	])
+}
+
+def off() {
+	log.debug "off()"
+	commands([
+		zwave.switchAllV1.switchAllOff(),
+		encap(zwave.switchBinaryV1.switchBinaryGet(), 1),
+		encap(zwave.switchBinaryV1.switchBinaryGet(), 2)
+	])
+}
+
+void childOn(String dni) {
+	log.debug "childOn($dni)"
+	def cmds = []
+	cmds << new physicalgraph.device.HubAction(command(encap(zwave.basicV1.basicSet(value: 0xFF), channelNumber(dni))))
+	cmds << new physicalgraph.device.HubAction(command(encap(zwave.switchBinaryV1.switchBinaryGet(), channelNumber(dni))))
+	sendHubCommand(cmds, 1000)
+}
+
+void childOff(String dni) {
+	log.debug "childOff($dni)"
+	def cmds = []
+	cmds << new physicalgraph.device.HubAction(command(encap(zwave.basicV1.basicSet(value: 0x00), channelNumber(dni))))
+	cmds << new physicalgraph.device.HubAction(command(encap(zwave.switchBinaryV1.switchBinaryGet(), channelNumber(dni))))
+	sendHubCommand(cmds, 1000)
+}
+
+void childRefresh(String dni) {
+	log.debug "childRefresh($dni)"
+	def cmds = []
+	cmds << new physicalgraph.device.HubAction(command(encap(zwave.switchBinaryV1.switchBinaryGet(), channelNumber(dni))))
+	sendHubCommand(cmds, 1000)
+}
+
+def poll() {
+	log.debug "poll()"
+	refresh()
+}
+
+def refresh() {
+	log.debug "refresh()"
+	commands([
+		encap(zwave.switchBinaryV1.switchBinaryGet(), 1),
+		encap(zwave.switchBinaryV1.switchBinaryGet(), 2),
+	])
+}
+
+def ping() {
+	log.debug "ping()"
+	refresh()
+}
+
+def getCheckInterval() {
+	2 * 15 * 60 + 2 * 60
+}
+
+def installed() {
+	log.debug "installed()"
+
+	sendEvent(name: "checkInterval", value: checkInterval, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
```

Comment:
this shouldn't be offlinepingable

Reply:
I removed offlinePingable from the children, but given the experience that @tpmanley had with the device not necessarily reporting physical events, wouldn't we want it as offlinePingable (since the device doesn't report metering).

That said, it existed in the original DTH (and in a few of the Z-Wave switches and dimmers), so I assume it was there for a reason.

Reply:
Alright, so I was confusing how `offlinePingable` is used today with how it was originally used... Since this and other switch devices were using the old definition, I've removed it.